docs/usage/webhook/deploy.rst

```diff
@@ -0,0 +1,152 @@
+------------------
+
+To create a new operation we first create a new Python package. DFFML has a script to do it for you.
+
+.. code-block:: console
+
+    $ dffml service dev create operations ffmpeg
+    $ cd ffmpeg
+
+Write operations and definitions to convert videos files to gif by calling
+``ffmpeg`` (Make sure you `download and install <https://www.ffmpeg.org/download.html>`_ it.).
+The operation accepts bytes (of the video), converts it to gif and outputs it as bytes.
+We will start writing our operation in ``./ffmpeg/operations.py``
+
+**ffmpeg/operations.py**
+
+.. literalinclude:: /../examples/ffmpeg/ffmpeg/operations.py
+
+Add the operation to ``ffmpeg/setup.py``
+
+.. code-block:: python
+
+    common.KWARGS["entry_points"] = {
+        "dffml.operation": [
+            f"convert_to_gif = {common.IMPORT_NAME}.operations:convert_to_gif"
+        ]
+    }
+
+Install the package
+
+.. code-block:: console
+
+    $ pip install -e .
+
+Dataflow and Config files
+-------------------------
+
+**ffmpeg/dataflow.py**
+
+.. literalinclude:: /../examples/ffmpeg/ffmpeg/dataflow.py
+
+.. code-block:: console
+
+    $ mkdir -p deploy/mc/http deploy/df
+    $ dffml service dev export -config yaml ffmpeg.dataflow:DATAFLOW > deploy/df/ffmpeg.yaml
+
+Create the config file for the HTTP service
+in ``deploy/mc/http/ffmpeg.yaml``
+
+.. code-block:: console
+
+    $ cat > ./deploy/mc/http/ffmpeg.yaml <<EOF
+    path: /ffmpeg
+    input_mode: bytes:input_file
+    output_mode: bytes:application/octet-stream:post_input.output_file
+    EOF
+
+- ``input_mode``
+
+    - ``bytes:input_file``
+
+        - We want the input from the request to be treated as bytes with definition ``input_file``.
+
+- ``output_mode``
+
+    - ``bytes:application/octet-stream:post_input.output_file``
+
+        - We want the response (content_type = application/octet-stream) to be bytes taken from  ``results["post_input"]["output_file"]``,
+          where ``results`` is the output of the dataflow.
+
+For more details see `HttpChannelConfig <../../plugins/service/http/dataflow.html#HttpChannelConfig>`__ .
+
+.. _usage_ffmpeg_deploy_serve:
+
+Serving the DataFlow
+--------------------
+
+Serving the dataflow on port 8080
+
+.. code-block:: console
+
+    $ dffml service http server -insecure -mc-config deploy -port 8080
+
+.. warning::
+
+    The ``-insecure`` flag is only being used here to speed up this
+    tutorial. See documentation on HTTP API
+    :doc:`/plugins/service/http/security` for more information.
+
+Now from another terminal, we can send post requests to the dataflow running at this port.
+
+.. code-block:: console
+
+    $ curl -v --request POST --data-binary @input.mp4 http://localhost:8080/ffmpeg -o output.gif
+
+You should replace ``input.mp4`` with path to your video file and ``output.gif`` to where you want the converted gif
+to be output to. An example video is available `here <https://github.com/intel/dffml/blob/master/examples/ffmpeg/>`_ .
```

Comment:
```suggestion
to be output to. An example video is available `here <https://github.com/intel/dffml/raw/master/examples/ffmpeg/tests/input.mp4>`_ .
```

Reply:
Below this line let's put a `.. warning` about how we've noticed that `snap install ffmpeg` did not work.